log-viewer/src/main/java/com/logviewer/data2/Log.java

```diff
@@ -13,28 +13,35 @@
 import org.springframework.lang.Nullable;
 
 import java.io.EOFException;
+import java.io.File;
+import java.io.FileOutputStream;
 import java.io.IOException;
 import java.nio.ByteBuffer;
 import java.nio.channels.SeekableByteChannel;
 import java.nio.charset.Charset;
-import java.nio.file.Files;
-import java.nio.file.NoSuchFileException;
-import java.nio.file.Path;
-import java.nio.file.StandardOpenOption;
+import java.nio.file.*;
 import java.nio.file.attribute.BasicFileAttributes;
 import java.security.MessageDigest;
 import java.security.NoSuchAlgorithmException;
+import java.util.Arrays;
+import java.util.List;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.ExecutorService;
 import java.util.function.Consumer;
 import java.util.function.Function;
 import java.util.function.Predicate;
 import java.util.zip.CRC32;
+import java.util.zip.GZIPInputStream;
 
 public class Log implements LogView {
 
     private static final Logger LOG = LoggerFactory.getLogger(Log.class);
 
+    private static final List<String> GZIP_CONTENT_TYPE = Arrays.asList(
+            "application/octet-stream", // gzip on linux!
```

Comment:
`application/octet-stream` may be any binary file, not only compressed using GZip: https://kb.iu.edu/d/agtj
Let's use the file extension to detect logs compressed by GZip